scipy/sparse/extract.py

```diff
@@ -35,11 +35,11 @@ def find(A):
 
     """
 
-    A = coo_matrix(A).tocsr()  # sums duplicates
-    A.eliminate_zeros()        # removes explicit zeros
-    A = A.tocoo(copy=False)    # (cheaply) convert to COO
-
-    return A.row,A.col,A.data
+    A = coo_matrix(A, copy=False)
```

Comment:
I'd happily merge this patch if the in-place canonicalizing were removed (and maybe turned into an issue for discussion).